### DISCUSSION TITLE
How to disable print_summary for failed fits so that multiprocessing in sklearn works properly?

### DISCUSSION THREAD

**Discussion by braza2** [CamDavidsonPilon/lifelines#1497] [Q&A]
Hi,
I have a problem, where after failed fits lifelines prints a summary. However, due to this printing, I can't properly make use of multiple cores while doing hyperparameter optimization with sklearn RandomizedSearchCV. Is there a way to stop any output to stdout, stderr?

**Comment by CamDavidsonPilon** [comment]
There may be a Python solution to redirect stdout, but I'm curious: what is being printed when a job fails? I should change it to put msgs in the Exception

**Reply by braza2** [reply]
this is whats being printed:
fun: 5.861624464464439
     jac: array([ 3.93387867e+01,  3.93504909e+01, -3.93529261e+01,  3.93670291e+01,
        3.93714435e+01, -3.93766713e+01,  3.93688176e+01, -3.93686001e+01,
        3.93475008e+01, -3.93490119e+01,  3.93482353e+01, -3.93626984e+01,
       -3.93482419e+01, -3.93571806e+01, -3.93585409e+01, -3.93734670e+01,
        3.93548911e+01, -3.93682726e+01,  3.93788482e+01, -3.93671020e+01,
       -3.93694288e+01, -3.93619260e+01,  3.93667550e+01,  3.93718513e+01,
       -3.93634317e+01,  3.93681456e+01,  3.93649871e+01, -3.93600469e+01,
        3.93644089e+01,  3.93646074e+01, -3.93639606e+01, -3.93608541e+01,
        3.93622978e+01,  3.93639474e+01,  3.93322415e+01, -3.93630667e+01,
        3.93657309e+01, -3.93617053e+01, -3.93638002e+01, -3.93604447e+01,
        3.93631177e+01, -3.93630613e+01, -3.93640295e+01, -3.93652665e+01,
        3.93622678e+01, -3.93786127e+01,  3.93649251e+01,  3.93621565e+01,
       -3.93621978e+01, -3.93645289e+01, -3.93666953e+01,  3.93639532e+01,
       -3.93644596e+01,  3.93632703e+01, -3.93644135e+01, -3.93649733e+01,
       -3.93677817e+01, -1.13118351e-19, -4.31482921e-20,  3.62534893e-20,
        3.93619336e+01, -3.93333428e+01,  3.93514694e+01, -8.45240318e-06,
       -4.03585195e+01])
 message: 'Iteration limit reached'
    nfev: 972
     nit: 200
    njev: 200
  status: 9
 success: False
       x: array([ 9.65258161e-09,  6.16444364e-13, -8.72473712e-09,  1.02487171e-08,
        1.91320057e-09, -6.83616903e-10,  2.09061628e-08, -1.90648710e-09,
        9.25061933e-09, -4.90861325e-09,  1.75946753e-08, -5.84909044e-09,
       -4.51751262e-09, -1.03953083e-08, -5.19810144e-08, -3.96772895e-09,
        3.24314105e-08, -5.40782871e-09,  3.15373745e-08, -6.78209815e-09,
       -1.93653019e-08, -1.67631780e-08,  2.20592047e-08,  1.25508421e-08,
       -1.33370431e-08,  1.90385159e-08,  9.96031242e-09, -9.41676723e-09,
        3.67683610e-09,  5.80928726e-08, -1.09179429e-08, -3.04628549e-09,
        2.21350954e-08,  1.06760401e-08,  4.21696341e-08, -4.25640836e-09,
        4.56001185e-08, -9.87410654e-10, -2.97194347e-09, -9.10692055e-09,
        1.33762742e-08, -3.21105167e-09, -3.64503163e-08, -1.33771157e-09,
        1.13152727e-08, -6.08104486e-09,  7.20558330e-10,  3.09522430e-09,
       -1.10457481e-09, -1.10123238e-08, -8.90643728e-08,  2.75688541e-09,
       -7.82266265e-09,  5.16274430e-09, -1.69033636e-08, -1.16367798e-08,
       -2.16425690e-08,  4.89472596e+00,  2.40464370e-16,  1.24827001e-15,
        4.19158215e-10, -2.98648128e-08,  2.28588282e-08,  4.86158026e+00,
       -3.67906030e-08])

**Comment by CamDavidsonPilon** [comment]
Ah, I do some printing when it fails. This information should be in the error message. https://github.com/CamDavidsonPilon/lifelines/blob/master/lifelines/fitters/__init__.py#L1963-L1966

There's currently no way to suppress this, you'd have to make a modification on your end
